src/colvarcomp.h

```diff
@@ -211,6 +209,27 @@ class colvar::cvc
   /// object
   virtual void apply_force(colvarvalue const &cvforce) = 0;
 
+  /// Whether the parameter param_name exists
+  virtual int param_exists(std::string const &param_name);
+
+  /// Value of the parameter param_name (must be a scalar)
+  virtual cvm::real get_param(std::string const &param_name);
+
+  /// Pointer to the parameter param_name (when it is an object)
+  virtual void const *get_param_ptr(std::string const &param_name);
+
+  /// Get a copy of the names of registered parameters
+  virtual std::vector<std::string> get_param_names();
+
+  /// Pointer to the gradients of the variable with respect to param_name
+  virtual colvarvalue const *get_param_grad(std::string const &param_name);
+
+  /// Get a copy of the names of registered parameter gradients
+  virtual std::vector<std::string> get_param_grad_names();
+
+  /// Set the named parameter to the given value
+  virtual int set_param(std::string const &param_name, void const *new_value);
+
```

Comment:
How about making `register_params` and associated functions members of `colvarparse`? That could be extremely generic, and I strongly agree that `register_params` can be bundled into `get_keyval`.

Admittedly `get_param_grad` is more CVC-specific, although it would work for a colvar, and possibly for a bias if that means the gradient of the bias energy.

Reply:
`get_param_grad` will be tricky, because its return value should be `colvarvalue *` for the one associated to a variable/CVC (gradient of a general quantity), and `real` for a bias (gradient of an energy), which I didn't add yet.  So these are two entirely different functions, both `virtual`...

To simplify a bit I moved the new functions to a their own class, which is for now inherited simply as `public` by `colvarparse` at the root, but we can play around with that if needed.